Update manifest: Nvidia.GeForceNow version 2.0.72.124

Dragon1573 opened the pull request:
## What's changed?

- Update installer URL and SHA256 hash value.

## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #244932

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

![image](https://github.com/user-attachments/assets/eca275cd-9d9c-4145-a437-57d17cc158d1)

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/245161)

Dragon1573 committed:
Update manifest: Nvidia.GeForceNow version 2.0.72.124

- Update installer URL and SHA256 hash value.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

SpecterShell commented:
Well this PR gonna change the InstallerUrl back to the vanity one...

Dragon1573 commented:
> Well this PR gonna change the `InstallerUrl` back to the vanity one...

Can't sure where static, version-related URL can be found. 😕 

The vanity URL was extracted from the official download site from `PackageUrl` in previous versions.

SpecterShell commented:
> > Well this PR gonna change the `InstallerUrl` back to the vanity one...
> 
> Can't sure where static, version-related URL can be found. 😕
> 
> The vanity URL was extracted from the official download site from `PackageUrl` in previous versions.

Then always check my automation code: https://github.com/SpecterShell/Dumplings/blob/main/Tasks/Nvidia.GeForceNow/Script.ps1

I guess I would revert this PR if it got merged anyways.

Dragon1573 commented on manifests/n/Nvidia/GeForceNow/2.0.72.124/Nvidia.GeForceNow.installer.yaml:
```diff
@@ -28,7 +28,7 @@ ProductCode: '{B2FE1952-0186-46C3-BAEC-A80AA35AC5B8}_GeforceNOW'
 Installers:
 - Architecture: x64
-  InstallerUrl: https://ota-downloads.nvidia.com/ota/GeForceNOW-release_E2BBA5.exe
+  InstallerUrl: https://download.nvidia.com/gfnpc/GeForceNOW-release.exe
```
```suggestion
  InstallerUrl: https://ota-downloads.nvidia.com/ota/GeForceNOW-release_E2BBA5.exe
```

Dragon1573 committed:
Apply suggestions from code review

- Roll back to static, version-related URL (inspected by @SpecterShell in Dumpling scripts)

Dragon1573 commented:
> Then always check [my automation code](https://github.com/SpecterShell/Dumplings/blob/main/Tasks/Nvidia.GeForceNow/Script.ps1) 🕵️

Successfully extracted (<https://github.com/microsoft/winget-pkgs/pull/245161/commits/3c7918fd7fc0af98d4b52cca18f45551a2c95b1b>). ❤️

Dragon1573 commented:
> [!IMPORTANT]
> 
> Seems @microsoft-github-policy-service[bot] can't automatically trigger the merge.
>
> @stephengillie, manual merged required. 🙏🏼
